pandas/tslib.pyx

```diff
@@ -1409,6 +1409,26 @@ cpdef inline object maybe_get_tz(object tz):
     return tz
 
 
+cpdef is_datetime_with_singletz_array(ndarray[object] values):
+    """
+    Check values have the same tzinfo attribute.
+    Doesn't check values are datetime-like types.
+    """
+    cdef Py_ssize_t i, n = len(values)
+    cdef object base_tz, tz
+
+    if n == 0:
+        return False
+
+    base_tz = getattr(values[0], 'tzinfo', None)
+    base_tz = get_timezone(base_tz)
+
+    for i in range(1, n):
+        tz = getattr(values[i], 'tzinfo', None)
+        if base_tz != get_timezone(tz):
+            return False
+    return True
```

Comment:
hmm ok

do an equality check on tz first then only check get_timezone if it's False (of course if that's False then return)

I think this would be much more performant